### PR TITLE
Cnp

### PR DESCRIPTION
**Pull request by mthava**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

**Test Configuration**:

* Hardware:
* O/S and version:
* JDK:

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: Jenkinsfile_CNP [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+def channel = '#div-dev'
+
+withPipeline(type , product, component) {
+    before('functionalTest:aat') {
+        env.TEST_ENVIRONMENT = "aat"
+    }
+
+    before('smokeTest:aat') {
+        error('Halting pipeline before prod stages')
+    }
```
Comment (mthava) [inline Jenkinsfile_CNP]: this should be after functionalTest:aat

File: Jenkinsfile_parameterized [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+                string(name: 'PRODUCT_NAME', defaultValue: 'div', description: 'Divorce'),
+                string(name: 'APP', defaultValue: 'document-generator', description: 'Divorce Document Generator Client'),
+                string(name: 'TYPE', defaultValue: 'java', description: ''),
+                string(name: 'ENVIRONMENT', defaultValue: 'saat', description: 'Environment where code should be built and deployed'),
+                choice(name: 'SUBSCRIPTION', choices: 'sandbox\nnonprod\nprod', description: 'Azure subscriptions available to build in')
+    ])
+])
+
+withParameterizedPipeline(params.TYPE, params.PRODUCT_NAME, params.APP, params.ENVIRONMENT, params.SUBSCRIPTION) {
+
```
Comment (mthava) [inline Jenkinsfile_parameterized]: Add the test_environment before functionalTest:aat

before('functionalTest:aat') {
+        env.TEST_ENVIRONMENT = "aat"
+    }

File: build.gradle [hunk 3]
```diff
@@ -1,124 +1,191 @@
...
-	id 'jacoco'
-	id 'pmd'
+    id 'application'
+    id 'checkstyle'
+    id 'com.github.ben-manes.versions' version '0.17.0'
+    id 'io.spring.dependency-management' version '1.0.5.RELEASE'
+    id 'jacoco'
+    id 'org.owasp.dependencycheck' version '3.1.2'
+    id 'org.sonarqube' version '2.6.2'
+    id 'org.springframework.boot' version'1.5.9.RELEASE'    // can update this?
```
Comment (mthava) [inline build.gradle]: remove this line

File: build.gradle [hunk 4]
```diff
@@ -1,124 +1,191 @@
...
 
 repositories {
-	mavenLocal()
-	jcenter()
+    mavenLocal()
+    jcenter()
+
+    maven { url "https://dl.bintray.com/hmcts/hmcts-maven" }
+    maven { url "http://repo.maven.apache.org/maven2" }
+    maven { url "https://repo.spring.io/libs-milestone" }
```
Comment (mthava) [inline build.gradle]: consider removing this

File: infrastructure/aat.tfvars [hunk 5]
```diff
@@ -0,0 +1,6 @@
+env = "aat"
+subscription = "nonprod"
+vault_env = "test"
+
+auth_provider_service_client_baseurl = "http://betaDevBccidamS2SLB.reform.hmcts.net:80"
+pdf_service_baseurl = "https://testpdfservicelb.moneyclaim.reform.hmcts.net:4302"
```
Comment (mthava) [inline infrastructure/aat.tfvars]: remove this line

File: infrastructure/demo.tfvars [hunk 6]
```diff
@@ -0,0 +1,6 @@
+env = "demo"
+subscription = "nonprod"
+vault_env = "preprod"
+
+auth_provider_service_client_baseurl = "https://preprod-s2s-api.reform.hmcts.net:3511"
+pdf_service_baseurl = "https://preprodpdfservicelb.moneyclaim.reform.hmcts.net:4301"
```
Comment (mthava) [inline infrastructure/demo.tfvars]: remove this line

File: infrastructure/main.tf [hunk 7]
```diff
@@ -0,0 +1,41 @@
+locals {
+  ase_name                           = "${data.terraform_remote_state.core_apps_compute.ase_name[0]}"
+  evidence_management_client_api_url = "http://${var.evidence_management_client_api_url_part}-${var.env}.service.${local.ase_name}.internal"
+
+  #tactical
+  #evidence_management_client_api_url = "http://betaDevBdivorceAppLB.reform.hmcts.net:4016"
```
Comment (mthava) [inline infrastructure/main.tf]: remove comments

File: infrastructure/main.tf [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+    REFORM_SERVICE_NAME                                   = "${var.reform_service_name}"
+    REFORM_TEAM                                           = "${var.reform_team}"
+    REFORM_ENVIRONMENT                                    = "${var.env}"
+    DIV_DOCUMENT_GENERATOR_PORT                           = "${var.div_document_generator_port}"
+    AUTH_PROVIDER_SERVICE_CLIENT_BASEURL                  = "${var.auth_provider_service_client_baseurl}"
+    AUTH_PROVIDER_SERVICE_CLIENT_MICROSERVICE             = "${var.auth_provider_service_client_microservice}"
+    AUTH_PROVIDER_SERVICE_CLIENT_KEY                      = "${data.vault_generic_secret.auth_provider_service_client_key.data["value"]}"
+    AUTH_PROVIDER_SERVICE_CLIENT_TOKENTIMETOLIVEINSECONDS = "${var.auth_provider_service_client_tokentimetoliveinseconds}"
+    //PDF_SERVICE_BASEURL                                   = "${local.pdf_service_url}"
+    PDF_SERVICE_BASEURL                                   = "${var.pdf_service_baseurl}"
```
Comment (mthava) [inline infrastructure/main.tf]: use local pdfurl. remove comment

File: infrastructure/preview.tfvars [hunk 9]
```diff
@@ -0,0 +1,6 @@
+env = "demo"
+subscription = "nonprod"
+vault_env = "preprod"
+
+auth_provider_service_client_baseurl = "https://preprod-s2s-api.reform.hmcts.net:3511"
+pdf_service_baseurl = "https://testpdfservicelb.moneyclaim.reform.hmcts.net:4302"
```
Comment (mthava) [inline infrastructure/preview.tfvars]: remove this line

File: infrastructure/prod.tfvars [hunk 10]
```diff
@@ -0,0 +1,6 @@
+env = "prod"
+subscription = "prod"
+vault_env = "prod"
+
+auth_provider_service_client_baseurl = "http://betaProdccidamAppLB.reform.hmcts.net:4502"
+pdf_service_baseurl = "https://prodpdfservicelb.moneyclaim.reform.hmcts.net:4301"
```
Comment (mthava) [inline infrastructure/prod.tfvars]: remove this line

File: infrastructure/saat.tfvars [hunk 11]
```diff
@@ -0,0 +1,6 @@
+env = "saat"
+subscription = "sandbox"
+vault_env = "test"
+
+auth_provider_service_client_baseurl = "http://betaDevBccidamS2SLB.reform.hmcts.net:80"
+pdf_service_baseurl = "https://testpdfservicelb.moneyclaim.reform.hmcts.net:4302"
```
Comment (mthava) [inline infrastructure/saat.tfvars]: remove this line

File: infrastructure/sandbox.tfvars [hunk 12]
```diff
@@ -0,0 +1,6 @@
+env = "sandbox"
+subscription = "sandbox"
+vault_env = "test"
+
+auth_provider_service_client_baseurl = "http://betaDevBccidamS2SLB.reform.hmcts.net:80"
+pdf_service_baseurl = "https://testpdfservicelb.moneyclaim.reform.hmcts.net:4302"
```
Comment (mthava) [inline infrastructure/sandbox.tfvars]: remove this line

File: infrastructure/variables.tf [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+
+variable "auth_provider_service_client_tokentimetoliveinseconds" {
+  default = "900"
+}
+
+variable "pdf_service_url_part" {
+  default = "cmc-pdf-service"
+}
+
+variable "pdf_service_baseurl" {
```
Comment (mthava) [inline infrastructure/variables.tf]: remove this

File: lib/AI-Agent.xml [hunk 14]
```diff
@@ -0,0 +1,24 @@
...
+            <Method name="storeDocumentAndGetInfo" reportCaughtExceptions="true" reportExecutionTime="true"/>
+        </Class>
+        <Class name="uk.gov.hmcts.reform.divorce.documentgenerator.service.impl.PDFGenerationServiceImpl">
+            <Method name="generateFromHtml" reportCaughtExceptions="true" reportExecutionTime="true"/>
+        </Class>
+        <Class name="uk.gov.hmcts.reform.divorce.documentgenerator.service.impl.TemplateManagementService">
+            <Method name="getTemplateByName" reportCaughtExceptions="true" reportExecutionTime="true"/>
+        </Class>
+    </Instrumentation>
+</ApplicationInsightsAgent>
```
Comment (mthava) [inline lib/AI-Agent.xml]: Review this and see if can find a way to log everything

File: src/integrationTest/java/uk/gov/hmcts/reform/divorce/ServiceContextConfiguration.java [hunk 15]
```diff
@@ -0,0 +1,46 @@
...
+        final ServiceAuthorisationApi serviceAuthorisationApi
+    ) {
+        return AuthTokenGeneratorFactory.createDefaultGenerator(secret, microService, serviceAuthorisationApi);
+    }
+
+    @Bean
+    public IDAMUtils getIDAMUtil() {
+        return new IDAMUtils();
+    }
+}
```
Comment (mthava) [inline src/integrationTest/java/uk/gov/hmcts/reform/divorce/ServiceContextConfiguration.java]: remove unnecessary resource files

File: src/integrationTest/resources/application-aat.properties [hunk 16]
```diff
@@ -0,0 +1,7 @@
+auth.idam.client.baseUrl=http://betaDevBccidamAppLB.reform.hmcts.net
+idam.s2s-auth.url=http://betaDevBccidamS2SLB.reform.hmcts.net:80
+
+idam.auth.secret=RJBNLKOV7OJR52VX
```
Comment (mthava) [inline src/integrationTest/resources/application-aat.properties]: remove this use the data from vault

File: src/integrationTest/resources/application-local.properties [hunk 17]
```diff
@@ -0,0 +1,22 @@
...
+#  Loggging                                   #
+###############################################
+logging.level.uk.gov.hmcts.ccd=DEBUG
+logging.level.org.springframework.web=DEBUG
+
+###############################################
+#  IDAM Auth S2S                              #
+###############################################
+auth.idam.client.baseUrl=http://betaDevBccidamAppLB.reform.hmcts.net
+#idam.auth.secret=RJBNLKOV7OJR52VX
```
Comment (mthava) [inline src/integrationTest/resources/application-local.properties]: remove

File: src/integrationTest/resources/application-local.properties [hunk 18]
```diff
@@ -0,0 +1,22 @@
...
+idam.auth.secret=AAAAAAAAAAAAAAAA
+#idam.auth.microservice=divorce_document_generator
```
Comment (mthava) [inline src/integrationTest/resources/application-local.properties]: remove

File: src/integrationTest/java/uk/gov/hmcts/reform/divorce/ServiceContextConfiguration.java (hunk 15 above)
Comment (johnoneill46) [reply src/integrationTest/java/uk/gov/hmcts/reform/divorce/ServiceContextConfiguration.java]: This has been addressed by deleting entire FileTypes folder

File: lib/AI-Agent.xml (hunk 14 above)
Comment (johnoneill46) [reply lib/AI-Agent.xml]: Story created into backlog for this to be done 
https://tools.hmcts.net/jira/browse/DIV-2524

**Pull request merged**